ddf-rpc-api/src/main/java/com/connexta/ddf/persistence/subscriptions/SubscriptionMethods.java

```diff
@@ -0,0 +1,145 @@
+package com.connexta.ddf.persistence.subscriptions;
+
+import com.connexta.jsonrpc.Error;
+import com.connexta.jsonrpc.MethodSet;
+import com.connexta.jsonrpc.RpcFactory;
+import com.connexta.jsonrpc.RpcMethod;
+import com.connexta.jsonrpc.email.EmailResolver;
+import com.connexta.jsonrpc.impl.RpcFactoryImpl;
+import com.connexta.util.MapFactory;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.persistence.PersistentStore.PersistenceType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SubscriptionMethods implements MethodSet {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionMethods.class);
+
+  private static final String EMAIL_PROPERTY = "emails";
+
+  private static final String SUBSCRIPTION_TYPE = PersistenceType.SUBSCRIPTION_TYPE.toString();
+
+  private static final Lock LOCK = new ReentrantLock();
+
+  private static final int START_INDEX = 0;
+
+  private static final int PAGE_SIZE = 1000;
+
+  private static final String ID = "id";
+
+  public static final String IDS = "ids";
+
+  private final PersistentStore persistentStore;
+
+  private final EmailResolver emailResolver;
+
+  private final Map<String, RpcMethod> METHODS;
+
+  private final RpcFactory rpc = new RpcFactoryImpl();
```

Comment:
✏️ Consider injecting this instead

Reply:
Hmm, so I probably put this here just out of laziness. Makes sense to pass it in though, particularly to help testing. Will change 👌